content/master/guides/composition-functions.md

```diff
@@ -0,0 +1,284 @@
+
+kubectl create namespace crossplane-system
+
+helm repo add crossplane-master https://charts.crossplane.io/master/
+helm repo update
+
+# Find the latest RC build of Crossplane - e.g. 1.11.0-rc.0.274.g51a07e50
+helm search repo crossplane-master/crossplane --devel
+
+RC_VERSION="v1.11.0-rc.0.274.g51a07e50"
+helm install crossplane --namespace crossplane-system crossplane-master/crossplane \
+    --devel --version ${RC_VERSION} \
+    --set "args={--debug,--enable-composition-functions}" \
+    --set "xfn.enabled=true" \
+    --set "xfn.args={--debug}"
+```
+
+The preceding Helm command installs Crossplane with the Composition Functions
+feature flag enabled, and with the reference "`xfn`" Composition Function runner
+deployed as a sidecar pod. You can confirm Composition Functions are enabled by
+looking for a log line like:
+
+```json
+{"level":"info","ts":1674535093.36186,"logger":"crossplane","msg":"Alpha feature enabled","flag":"EnableAlphaCompositionFunctions"}
+```
+
+{{< hint "tip" >}}
+Use `kubectl -n crossplane-system logs -l app=crossplane` to see the logs. You
+should see the "Alpha feature enabled" log line emitted shortly after Crossplane
+starts.
+{{< /hint >}}
+
+## The xfn Runner
+
+Composition Function runners are designed to be pluggable. Each time Crossplane
+needs to invoke a Composition Function it makes a gRPC call to a configurable
+endpoint. The default, reference Composition Function runner is named `xfn`.
+
+{{< img src="composition-functions-xfn-runner.png" alt="Crossplane running functions using xfn via gRPC" size="tiny" >}}
+
+`xfn` is deployed as a sidecar container within the Crossplane pod. It runs each
+Composition Function as a nested [rootless container][rootless-containers]. The
+Crossplane Helm chart deploys `xfn` with slightly elevated permissions. Namely:
+
+* The [Unconfined seccomp profile][kubernetes-seccomp].
+* `CAP_SETUID` and `CAP_SETGID`.
+
+`xfn` doesn't strictly need either of these. It's possible to run it under
+seccomp but doing so is onerous. You must author a custom profile that allows a
+few additional syscalls required to run containers, including `unshare`,
+`mount`, and `unmount`. Refer to the [seccomp documentation][kubernetes-seccomp]
+for information on how to do so.
+
+Granting `CAP_SETUID` and `CAP_SETGID` is optional. Doing so allows `xfn` to
+create slightly better rootless containers. If `xfn` is not granted these
+capabilities it will not support multiple users and groups inside Composition
+Function containers; only UID and GID 0 will exist. Regardless of capabilities
+`xfn` always runs each Composition Function as an unprivileged user. That user
+will appear to be root inside the Composition Function container thanks to 
+[`user_namespaces(7)`].
+
+## Using Composition Functions
+
+To use Composition Functions you must:
+
+1. Find one or more Composition Functions, or write your own.
```

Comment:
We don't have a good discovery mechanism at the moment. I might update this with a link to https://github.com/crossplane/crossplane/discussions suggesting that folks share their functions there.